financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/DefaultEventNotificationGenerator.java

```diff
@@ -0,0 +1,85 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.financial.services.accelerator.event.notifications.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.financial.services.accelerator.common.config.FinancialServicesConfigParser;
+import org.wso2.financial.services.accelerator.common.util.Generated;
+import org.wso2.financial.services.accelerator.common.util.JWTUtils;
+import org.wso2.financial.services.accelerator.event.notifications.service.exception.FSEventNotificationException;
+import org.wso2.financial.services.accelerator.event.notifications.service.model.Notification;
+import org.wso2.financial.services.accelerator.event.notifications.service.model.NotificationEvent;
+import org.wso2.financial.services.accelerator.event.notifications.service.model.NotificationResponse;
+import org.wso2.financial.services.accelerator.event.notifications.service.util.EventNotificationServiceUtil;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Default Event Notification Response Generator Class.
+ */
+public class DefaultEventNotificationGenerator implements EventNotificationGenerator {
+
+    private static Log log = LogFactory.getLog(DefaultEventNotificationGenerator.class);
+
+    @Override
+    public NotificationResponse generateEventNotificationBody(Notification notification,
+                                                              List<NotificationEvent> notificationEventList)
+            throws FSEventNotificationException {
+
+        NotificationResponse notificationResponse = new NotificationResponse();
+        //get current time in milliseconds
+        Long currentTime = Instant.now().getEpochSecond();
+
+        //generate transaction Identifier
+        String transactionIdentifier = UUID.randomUUID().toString();
+
+        notificationResponse.setIss(FinancialServicesConfigParser.getInstance().getEventNotificationTokenIssuer());
+        notificationResponse.setIat(currentTime);
+        notificationResponse.setAud(notification.getClientId());
+        notificationResponse.setJti(notification.getNotificationId());
+        notificationResponse.setTxn(transactionIdentifier);
+        notificationResponse.setToe(notification.getUpdatedTimeStamp());
+        notificationResponse.setSub(generateSubClaim(notification));
+        notificationResponse.setEvents(notificationEventList);
+        return notificationResponse;
+    }
+
+    @Generated(message = "Excluded from tests as using a util method from a different package")
+    public String generateEventNotification(JsonNode jsonNode) throws FSEventNotificationException {
+
+        String payload = EventNotificationServiceUtil.getCustomNotificationPayload(jsonNode);
+        try {
+            return JWTUtils.signJWTWithDefaultKey(payload);
+        } catch (Exception e) {
```

Comment:
Need to catch a specific exception

Reply:
To get the default key of the server, I am using keyStoreManager.getDefaultPrivateKey() in IS and that method throws an Exception. That's why I am catching that here